### PR TITLE
Kamu UI 314 mqtt metadata block support

### PR DESCRIPTION
**Pull request by dmitriy-borzenko**

### CODE REVIEW

**Comment by dmitriy-borzenko** [comment]
Result: Changed lineage graph

image

File: src/app/dataset-view/additional-components/lineage-component/services/lineage-graph-builder.service.ts [hunk 1]
```diff
@@ -105,40 +105,78 @@ export class LineageGraphBuilderService {
...
+                    sourceNodesByLabel.set(sourceNodeLabel, sourceNode);
+                }
+
+                source2DatasetLinks.push({
+                    id: `${sourceNode.id}__and__${datasetId}`,
+                    source: sourceNode.id,
+                    target: datasetId,
+                } as Edge);
+            });
+        }
```
Comment (zaychenko-sergei) [inline src/app/dataset-view/additional-components/lineage-component/services/lineage-graph-builder.service.ts:179]: Large amount of duplication. Isolate functions with arguments for `kind`, ...

**Review by zaychenko-sergei — CHANGES_REQUESTED** [review]
No tests?

File: src/app/dataset-view/additional-components/lineage-component/services/lineage-graph-builder.service.ts (hunk 1 above)
Comment (dmitriy-borzenko) [reply src/app/dataset-view/additional-components/lineage-component/services/lineage-graph-builder.service.ts:179]: Moved to separate function

**Review by zaychenko-sergei — APPROVED** [review]

**Pull request merged**